src/main/scala/uk/gov/nationalarchives/tdr/GraphQLClient.scala

```diff
@@ -23,41 +24,67 @@ class GraphQLClient[Data, Variables](url: String)(implicit val ec: ExecutionCont
 
   implicit val customConfig: Configuration = Configuration.default.withDefaults
 
-  case class GraphqlData(data: Option[Data], errors: List[GraphqlError] = Nil)
+  case class GraphqlData(data: Option[Data], errors: List[Error] = Nil)
 
-  def getResult(token: BearerAccessToken, document: Document, variables: Option[Variables] = Option.empty): Future[GraphqlData] = {
+  def getResult(token: BearerAccessToken, document: Document, variables: Option[Variables] = Option.empty): Future[GraphQlResponse[Data]] = {
     val queryJson: Json = Json.fromString(QueryRenderer.render(document, QueryRenderer.Compact))
     val variablesJson: Option[Json] = variables.map(_.asJson)
     val fields: immutable.Seq[(String, Json)] = List("query" -> queryJson) ++ variablesJson.map("variables" -> _)
 
     val body = Json.obj(fields: _*).printWith(GraphQLClient.jsonPrinter)
 
-    val response: Future[Response[Either[ResponseError[Error], GraphqlData]]] =
+    val response: Future[Response[Either[String, String]]] =
       basicRequest
         .post(uri"$url")
         .auth.bearer(token.getValue)
         .body(body)
         .contentType(MediaType.ApplicationJson)
-        .response(asJson[GraphqlData]).send()
+        .response(asString)
+        .send()
 
-    response.map(r => {
-      r.body match {
-        case Right(r) => r
-        case Left(e) => GraphqlData(Option.empty, List(GraphqlError(e.body, List(), List())))
+    response.flatMap(r => {
+      if (r.isSuccess) {
+        r.body match {
+          case Right(body) => parseBody(body)
+          case Left(error) => Future.failed(new GraphQlClientException(error))
+        }
```

Comment:
Do we need the `r.isSuccess` and the match of left and right. From https://sttp.softwaremill.com/en/latest/responses/body.html a Left on the Either means that the response code is not 20x but if isSuccess is true then the response code is 20x. Can this be simplified by removing the if and keeping the match

Reply:
Yeah, that makes sense. 👍 

I think I added the if statement before I split out the JSON parsing, and it kind of made sense to distinguish 200 errors from parsing errors. But now the JSON parsing is handled completely separately, so I've done what you suggested. We still have the full response body in the exception, so the client should have the right information available if we need to debug this error.